### DISCUSSION TITLE
How to add playground while using APIHandler?

### DISCUSSION THREAD

**Discussion by mesutgurlek** [langchain-ai/langserve#467] [Q&A] [answered]
I am trying similar use case with https://github.com/langchain-ai/langserve/blob/main/examples/api_handler_examples/server.py 
Especially working with the "Example 2". Playground url does not work and the swagger ui does not allow input json field. However I am able to call api from postman.
Am I missing something, could not find explanation in the docs or examples.
Thanks!

**Comment by eyurtsev** [comment, marked as answer]
When you use the APIHandler you are responsible for creating the endpoints. It provides flexibility to control the request path completely, but at the cost of extra work to create the endpoints manually.


`add_routes` is a wrapper around APIHandler to automatically register all the needed endpoints. 

https://github.com/langchain-ai/langserve/blob/main/langserve/server.py#L231

Take a look at the code to see how it registers the playground endpoint.

Make sure to assign a correct path to the runnable when using APIHandler since that's needed to server playground assets.

**Reply by mesutgurlek** [reply]
Thanks
